funcbench/README.md

```diff
@@ -1,100 +1,33 @@
 # funcbench
 
-A tool used as a github action to run a `go test -bench` and compare changes from a PR against another branch or commit.
+Benchmark and compare your Go code between commits or sub benchmarks. It uses `go test -bench` to run the benchmarks and uses [benchcmp](https://godoc.org/golang.org/x/tools/cmd/benchcmp) to compare them.
 
-The benchmark can be triggered by creating a comment which specifies a branch to compare. The results are then posted back as a PR comment.
-The benchmark can be also trigger and used as CLI command, without GitHub hook.
+funcbench currently supports two modes, Local and GitHub. Running it in the Github Mode also allows it to accept *a pull request number* and *a branch/commit* to compare against, which makes it suitable for automated tests.
 
-Comparison is done using [benchcmp](https://godoc.org/golang.org/x/tools/cmd/benchcmp).
-Arguments for the benchcmp are read from files created by previous action (for example [commentMonitor](/tools/commentMonitor/main.go)),
-which is responsible for the comment parsing.
+### Environment variables
+- `GITHUB_TOKEN`: Access token to post benchmarks results to respective PR.
 
-## Usage
+## Usage Examples
+> Clean git state is required.
 
-NOTE: Clean git state is required.
-Examples:
-
-* Execute benchmark named `FuncName` regex, and compare it with `master` branch.
-
- ```
- /funcbench -v master BenchmarkFuncName
- ```
-
-* Execute all benchmarks matching `FuncName.*` regex, and compare it with `master` branch.
-
-```
- /funcbench -v master FuncName.*
- ```
-
-* Execute all benchmarks, and compare the results with `devel` branch.
-
- ```
- /funcbench -v devel .
- ```
-
-* Execute all benchmarks matching `FuncName.*` regex, and compare it with `6d280faa16bfca1f26fa426d863afbb564c063d1` commit.
-
- ```
- /funcbench -v 6d280faa16bfca1f26fa426d863afbb564c063d1 FuncName.*
- ```
-
-* Execute all benchmarks matching `FuncName.*` regex on current code. Compare it between sub-benchmarks (`b.Run`) of same benchmark for current commit.
-Errors out if there are no sub-benchmarks.
-
- ```
- /funcbench -v . FuncName.*
- ```
-
-### GitHub
-
-Tests are triggered by posting a comment in a PR with the following format:
-
-`/funcbench <branch/commit> <Go test regex>`
-
-Specifying which tests to run are filtered by using the standard [Go regex RE2 language](https://github.com/google/re2/wiki/Syntax).
-
-* To test it locally, set `-w` flag or `WORKSPACE` environment variable to an empty directory where the source will be cloned.
-
-By default all benchmarks run without `-race` flag (#275).
-
-#### Example Github actions workflow file to pass in --input flag.
-
-> TODO: No longer using `issue_comment`, to be replaced with commentMonitor usage.
+|Usage|Command|
```

Comment:
This is nice. But what I would recommend is:

* Put all of this into funcbench --help in code.
* Use embedmd as we do in Thanos to automatically generate docs from it! (: See: https://github.com/thanos-io/thanos/blob/cc0b7e62b8b2c6f0879e5bfdd9881a1f7d699fc4/scripts/genflagdocs.sh#L3

This way we can have only one place that we need to copy this and update on changes + it is available on `funcbench --help! :hugs:

Reply:
What do you think?

Reply:
Hm?

Reply:
@krasi-georgiev mentioned doing this with prombench and with the other tools almost a year ago. I'll create an issue for this so we can work on this. :)

Reply:
should we put the usage info into the help command in this PR?

Reply:
Nah, we can do later

Reply:
Can we add TODO/Github issue?